src/ReaderPluginManager.php

```diff
@@ -26,25 +29,9 @@ class ReaderPluginManager extends AbstractPluginManager
         'javaproperties'  => 'Zend\Config\Reader\JavaProperties',
     ];
 
-    /**
-     * Validate the plugin
-     * Checks that the reader loaded is an instance of Reader\ReaderInterface.
-     *
-     * @param  Reader\ReaderInterface $plugin
-     * @return void
-     * @throws Exception\InvalidArgumentException if invalid
-     */
-    public function validatePlugin($plugin)
+    public function __construct(ContainerInterface $container, array $config = [])
     {
-        if ($plugin instanceof Reader\ReaderInterface) {
-            // we're okay
-            return;
-        }
-
-        throw new Exception\InvalidArgumentException(sprintf(
-            'Plugin of type %s is invalid; must implement %s\Reader\ReaderInterface',
-            (is_object($plugin) ? get_class($plugin) : gettype($plugin)),
-            __NAMESPACE__
-        ));
+        $config = array_merge_recursive(['invokables' => $this->invokableClasses], $config);
```

Comment:
This approach is necessary as the SM doesn't have an `$invokableClasses` property any longer; it instead munges invokables into factory and alias definitions. As such, the configuration needs to be passed in during instantiation.